libcsm/sls/get_xname.py

```diff
@@ -0,0 +1,49 @@
+#
+#  MIT License
+#
+#  (C) Copyright 2023 Hewlett Packard Enterprise Development LP
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a
+#  copy of this software and associated documentation files (the "Software"),
+#  to deal in the Software without restriction, including without limitation
+#  the rights to use, copy, modify, merge, publish, distribute, sublicense,
+#  and/or sell copies of the Software, and to permit persons to whom the
+#  Software is furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included
+#  in all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+#  THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+#  OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+#  ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+#  OTHER DEALINGS IN THE SOFTWARE.
+#
+"""
+Function for setting boot-image in BSS
+"""
+
+import sys
+
+from argparse import ArgumentParser
+from libcsm.sls import api
+
+def main():
+
+    # parse arguments
+    parser = ArgumentParser(description='Set BSS image utility.')
```

Comment:
I'd encourage you to use `click` instead of `argparse`, it saves you some time.

Reply:
For some examples of `click`, see:
- newer: https://github.com/Cray-HPE/crucible/blob/main/crucible/cli.py
- older: https://github.com/Cray-HPE/craycli/blob/master/cray/cli.py

Note that you can use click in multiple files, the examples use a singular file for the entire app.

Reply:
I did implement click here